creator-node/src/ipfsAdd.js

```diff
@@ -0,0 +1,255 @@
+
+  const resps = []
+  for await (const file of importer(content, block, options)) {
+    resps.push({
+      path: file.path,
+      cid: `${file.cid}`,
+      size: file.size
+    })
+  }
+
+  // Note: According to https://github.com/ipfs/js-ipfs-unixfs/tree/master/packages/ipfs-unixfs-importer#example,
+  // the importer will return the root as the last file resp. This means that the dir should always be the last index.
+  // (As we need it to be in resizeImage.js)
+  return resps
+}
+
+/**
+ * Wrapper for ipfs.add() with flag to exclusively generate the CID using the only hash logic, or to also add content to the ipfs daemon.
+ * Used for adding non-images (track segments, track transcode, metadata).
+ * @param {Object} ipfsLatest ipfs instance (should be v43.0.1)
+ * @param {Buffer|ReadStream|string} content a single Buffer, a ReadStream, or path to an existing file
+ * @param {Object?} ipfsConfig ipfs add config options
+ * @param {Object?} logContext
+ * @param {boolean?} enableIPFSAdd flag to add content to ipfs daemon
+ * @returns {string} only hash response cid or ipfs daemon response cid
+ */
+ipfsAdd.ipfsAddNonImages = async (content, ipfsConfig = {}, logContext = {}, enableIPFSAdd = false) => {
+  const logger = genericLogger.child(logContext)
+
+  let buffer = await _convertToBuffer(content, logger)
+
+  const startOnlyHash = hrtime.bigint()
+  const onlyHash = await ipfsAdd.ipfsOnlyHashNonImages(buffer)
+  const durationOnlyHashMs = (hrtime.bigint() - startOnlyHash) / BigInt(1000000) // convert ns -> ms
+
+  if (!enableIPFSAdd) {
+    logger.info(`[ipfsClient - ipfsAddNonImages()] onlyHash=${onlyHash} onlyHashDuration=${durationOnlyHashMs}ms`)
+    return onlyHash
+  }
+
+  /* ipfs.add with the v43.0.1 (aka ipfsLatest.add) returns a async iterator. To access this response, use a `for await... of` loop. Then,
+      the response will have the structure:
+      {
+        path: 'docs/assets/anchor.js',
+        cid: CID('QmVHxRocoWgUChLEvfEyDuuD6qJ4PhdDL2dTLcpUy3dSC2'),
+        size: 15347
+      }
+      See https://www.npmjs.com/package/ipfs-http-client/v/43.0.1#example
+  */
+  try {
+    let ipfsDaemonHash
+    if (!ipfsConfig.timeout) {
+      ipfsConfig.timeout = IPFS_ADD_TIMEOUT_MS
+    }
+
+    const startIpfsLatestAdd = hrtime.bigint()
+    for await (const ipfsLatestAddWithDaemonResp of ipfsLatest.add(buffer, ipfsConfig)) {
+      ipfsDaemonHash = `${ipfsLatestAddWithDaemonResp.cid}`
+    }
+    const durationIpfsLatestAddMs = (hrtime.bigint() - startIpfsLatestAdd) / BigInt(1000000) // convert ns -> ms
+
+    const isSameHash = onlyHash === ipfsDaemonHash
+    logger.info(`[ipfsClient - ipfsAddNonImages()] onlyHash=${onlyHash} onlyHashDuration=${durationOnlyHashMs}ms ipfsDaemonHash=${ipfsDaemonHash} ipfsDaemonHashDuration=${durationIpfsLatestAddMs}ms isSameHash=${isSameHash}`)
+
+    if (!isSameHash) {
+      throw new Error(`onlyHash=${onlyHash} and ipfsDaemonHash=${ipfsDaemonHash} are not consistent`)
+    }
```

Comment:
how do you feel about restructuring this a bit ie
```
    const isSameHash = onlyHash === ipfsDaemonHash
    if (isSameHash) {
      logger.info(`[ipfsClient - ipfsAddNonImages()] onlyHash=${onlyHash} onlyHashDuration=${durationOnlyHashMs}ms ipfsDaemonHash=${ipfsDaemonHash} ipfsDaemonHashDuration=${durationIpfsLatestAddMs}ms isSameHash=${isSameHash}`)
    } else {
      logger.error(`[ipfsClient - ipfsAddNonImages()] hash mismatch: onlyHash=${onlyHash} onlyHashDuration=${durationOnlyHashMs}ms ipfsDaemonHash=${ipfsDaemonHash} ipfsDaemonHashDuration=${durationIpfsLatestAddMs}ms`)
      throw new Error(`[ipfsClient - ipfsAddNonImages()] hash mismatch: onlyHash=${onlyHash} onlyHashDuration=${durationOnlyHashMs}ms ipfsDaemonHash=${ipfsDaemonHash} ipfsDaemonHashDuration=${durationIpfsLatestAddMs}ms`)
    }
```

Reply:
few reasons
1) current code will info log even if `isSameHash == false` which i've swapped to error log
2) if `isSameHash == false` currently there is no log and only an error thrown, so i added an error log
3) i noticed the thrown error msg has an inconsistent format with no prefix `[ipfsClient - ipfsAddNonImages()]` which i added
4) this logically seems more consistent and readable as an if-else
5) felt `[ipfsClient - ipfsAddNonImages()] hash mismatch:` was an easier string to log filter for as opposed to current which has the injected fields at the beginning and the filterable string literal is `are not consistent` which is much less specific

Reply:
tbh, I didn't particularly want to add conditionals for logging statements because I think it adds a little extra complexity that can be expressed in a try/catch block. I also added this errMsg in the catch block to log the err msg which has the prefix `[ipfsClient - ipfsAddNonImages()]` 😃 

```
// code ... 
    if (!isSameHash) {
      throw new Error(`onlyHash=${onlyHash} and ipfsDaemonHash=${ipfsDaemonHash} are not consistent`)
    } catch (e) {
    const errorMsg = `[ipfsClient - ipfsAddNonImages()] Could not add content to ipfs: ${e.toString()}`
    logger.error(errorMsg)
    throw new Error(errorMsg)
  }
```

also, I was thinking about filtering on `isSameHash=false` and `isSameHash=true` if we are to add a loggly dashboard to track this log. I think tracking on variable value instead of error message is more sustainable as error msgs are more likely prone to changes in the future.

Reply:
ah ya i missed the catch block error log - that invalidates comments 2,3,5. not a fan of info logging isSameHash=false but its nbd, resolving